firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -177,20 +180,27 @@ public InstallationResponse createFirebaseInstallation(
   private void writeFIDCreateRequestBodyToOutputStream(
       HttpURLConnection httpURLConnection, @NonNull String fid, @NonNull String appId)
       throws IOException {
-    OutputStream outputStream = httpURLConnection.getOutputStream();
-    if (outputStream == null) {
-      throw new IOException(
-          "Cannot send CreateInstallation request to FIS. No OutputStream available.");
-    }
-
-    GZIPOutputStream gzipOutputStream = new GZIPOutputStream(outputStream);
     try {
-      gzipOutputStream.write(
-          buildCreateFirebaseInstallationRequestBody(fid, appId).toString().getBytes("UTF-8"));
+      writeRequestBodyToOutputStream(
+          httpURLConnection, getJsonBytes(buildCreateFirebaseInstallationRequestBody(fid, appId)));
     } catch (JSONException e) {
       throw new IllegalStateException(e);
```

Comment:
Where is the `JSONException` thrown? How about catching this `JSONException` where it is thrown and throw an IllegalStateException instead. That way this method even becomes cleaner and we have less code duplication (of catching this exception).
If you do, please add in JavaDoc of this comment that it might throw an `IllegalStateException` in case there is a problem with the JSON.

Reply:
Done.